hamilton/plugins/sklearn_plot_extensions.py

```diff
@@ -0,0 +1,53 @@
+import dataclasses
+from os import PathLike
+from typing import Any, Collection, Dict, Tuple, Type, Union
+
+
+try:
+    import sklearn.metrics
+except ImportError:
+    raise NotImplementedError("scikit-learn is not installed.")
+
+try:
+    from sklearn.exceptions import NotFittedError
+except ImportError:
+    raise NotImplementedError("scikit-learn is not installed.")
+
+try:
+    from matplotlib import pyplot as plt
+except ImportError:
+    raise NotImplementedError("matplotlib is not installed")
+
+from hamilton import registry
+from hamilton.io import utils
+from hamilton.io.data_adapters import DataSaver
+
+
+SKLEARN_PLOT_TYPES = [sklearn.metrics.ConfusionMatrixDisplay, sklearn.metrics.DetCurveDisplay, sklearn.metrics.PrecisionRecallDisplay, sklearn.metrics.PredictionErrorDisplay, sklearn.metrics.RocCurveDisplay]
+SKLEARN_PLOT_TYPES_ANNOTATION = Union[sklearn.metrics.ConfusionMatrixDisplay, sklearn.metrics.DetCurveDisplay, sklearn.metrics.PrecisionRecallDisplay, sklearn.metrics.PredictionErrorDisplay, sklearn.metrics.RocCurveDisplay]
+
+@dataclasses.dataclass
+class ConfusionMatrixPlotSaver(DataSaver):

```

Comment:
We should rename this to something more generic, since it handles different sklearn displays.

Reply:
Changing it to `SklearnPlotSaver`